elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/metric/BaseMetric.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.datastores.aggregation.metric;
+
+import com.yahoo.elide.datastores.aggregation.Column;
+import com.yahoo.elide.datastores.aggregation.annotation.Meta;
+import com.yahoo.elide.datastores.aggregation.annotation.MetricAggregation;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringJoiner;
+import java.util.stream.Collectors;
+
+/**
+ * {@link Metric} annotated by {@link MetricAggregation}.
+ * <p>
+ * {@link BaseMetric} is thread-safe and can be accessed by multiple threads.
+ */
+@Slf4j
+public class BaseMetric extends Column implements Metric {
```

Comment:
Can we rename BaseMetric to AggregatedMetric?

Reply:
Sure